### ISSUE TITLE
Chat notifications are needlessly duplicated

### ISSUE THREAD

**Issue by orlp** [runelite/runelite#12187]
**Describe the bug**
When adding a highlighted word to the "Chat Notifications" plugin and notifying on a highlighted word, the whole message gets duplicated. When the original message was already highlighted, this is quite annoying. For example:
![image](https://i.imgur.com/HBqMi12.png)

This happens also for "Untradeable drop" or "You have a funny feeling you're being followed", which are other examples of red messages. Yet I would like to still add these because I want to be notified when this happens.

**To Reproduce**
1. Add a word to the "Highlight words" list that occurs in the game already red.
2. Enable "notify on highlight".
3. Trigger an event that causes such a message. Easiest way is to put your 'valuable item filter' in-game to 1 gp.
4. Note the message is duplicated.

**Expected behavior**
The message is only shown once in the chat log.

**Proposed solution**
`runelite.client.Notifier.notify` should get another argument, `shouldAddChatMessage`, defaulting to true. Then in the chat notification plugin it should check if the message is already red and highlighted. If yes it should call the `notify` function with the `shouldAddChatMessage` argument set to false.

**Comment by Broooklyn** [comment]
This is caused by `enable game message notifications` in `RuneLite`

**Comment by orlp** [comment]
@Broooklyn You're right, let me add that to the reproduction steps. Regardless, it should not duplicate the message even with that option turned on (because I do like having all other notifications as a chat message as well).

**Comment by orlp** [comment]
An alternative solution (which I'm not sure is possible with the API) is that `runelite.client.Notifier.notify` checks if its argument equals any of the last K messages (for some small K to beat race conditions, K=1 if those are not possible) in the chat, and if so edits it to make it red (if not already red) as opposed to adding a new red message.
